### PR TITLE
Migrate to Spring 6, Spring Security 6, Tomcat 10 

### PR DESCRIPTION
**Pull request by vkalapov**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,26 +20,27 @@
...
+        <spring.version>6.1.10</spring.version>
+        <spring-security.version>6.3.1</spring-security.version>
         <json-smart.version>2.5.1</json-smart.version>
         <hikari.version>5.1.0</hikari.version>
         <jackson.version>2.17.2</jackson.version>
         <jackson.databind.version>2.17.2</jackson.databind.version>
         <liquibase.version>4.29.1</liquibase.version>
         <liquibase-slf4j.version>5.0.0</liquibase-slf4j.version>
-        <cloudfoundry-client.version>2.47.0</cloudfoundry-client.version>
+        <cloudfoundry-client.version>2.48.0-SNAPSHOT</cloudfoundry-client.version>
```
Comment (ikasarov) [inline pom.xml:43]: this probably needs to be changed to a specific released version before merging

File: pom.xml [hunk 2]
```diff
@@ -49,13 +50,13 @@
         <aliyun-sdk-oss.version>3.17.4</aliyun-sdk-oss.version>
         <auto-service.version>1.1.1</auto-service.version>
         <commons-codec.version>1.16.1</commons-codec.version>
-        <multiapps.version>2.24.0</multiapps.version>
+        <multiapps.version>2.25.0-SNAPSHOT</multiapps.version>
```
Comment (ikasarov) [inline pom.xml:53]: same here

**Review by ikasarov — DISMISSED** [review]

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CustomControllerClientErrorHandler.java [hunk 3]
```diff
@@ -48,7 +48,8 @@ protected ResilientCloudOperationExecutor createExecutor(HttpStatus... statusesT
 
     private CloudOperationException asCloudOperationException(HttpStatusCodeException exception) {
         String description = getDescriptionFromResponseBody(exception.getResponseBodyAsString());
-        return new CloudOperationException(exception.getStatusCode(), exception.getStatusText(), description);
+        HttpStatus httpStatus =  HttpStatus.valueOf(exception.getStatusCode().value());
```
[the remark this answers is not in the document]
Comment (vkalapov) [reply multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CustomControllerClientErrorHandler.java]: Formatted it in the latest changes.

File: multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLogsPersistenceService.java [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
+import org.springframework.context.annotation.Primary;
+
+import java.sql.SQLException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Named("processLogsPersistenceService")
+@Primary
```
[the remark this answers is not in the document]
Comment (vkalapov) [reply multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLogsPersistenceService.java]: Leftover from development testing. Removed

File: multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProgressMessageService.java [hunk 5]
```diff
@@ -19,6 +19,7 @@
 import org.springframework.context.annotation.Primary;
 
 @Named
+@Primary
```
[the remark this answers is not in the document]
Comment (vkalapov) [reply multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProgressMessageService.java]: Leftover from development testing. Removed

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/FileUploadThreadPoolConfiguration.java [hunk 6]
```diff
@@ -1,19 +1,12 @@
...
-import java.util.concurrent.TimeUnit;
-
-import javax.inject.Inject;
-
+import jakarta.inject.Inject;
 import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.util.concurrent.*;
```
[the remark this answers is not in the document]
Comment (vkalapov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/FileUploadThreadPoolConfiguration.java]: Fixed. Problem came from a configuration in the IDE for auto imports.

File: pom.xml [hunk 7]
```diff
@@ -20,26 +20,27 @@
...
         <httpclient.version>4.5.14</httpclient.version>
         <httpcore.version>4.4.16</httpcore.version>
-        <eclipselink.version>2.7.14</eclipselink.version>
+        <eclipselink.version>3.0.4
+        </eclipselink.version>  <!-- Be careful when updating eclipselink versions. As of version 4.0.4, there have been reported regressions connected to exhausted Hikari DB pools  -->
         <flowable.version>6.8.0</flowable.version>
         <mybatis.version>3.5.16</mybatis.version>
-        <spring.version>5.3.39</spring.version>
-        <spring-security.version>5.8.11</spring-security.version>
+        <spring.version>6.1.10</spring.version>
```
[the remark this answers is not in the document]
Comment (vkalapov) [reply pom.xml]: Changed to latest versions.

**Review by IvanBorislavovDimitrov — DISMISSED** [review]

**Review by IvanBorislavovDimitrov — APPROVED** [review]

**Pull request merged**